File: n657-judge-route-circle.js
var judgeCircle = function(moves) {
    let x = 0
    let y = 0

    for (let i = 0; i < moves.length; i++) {
        switch(moves[i]) {
            case 'R':
                x++;
                break;
            case 'L':
                x--;
                break;
            case 'U':
                y++;
                break;
            case 'D':
                y--;
                break;
        }
    }

    return x === 0 && y === 0
}


console.log(judgeCircle("RULD"))
console.log(judgeCircle("RULR"))

File: n657-judge-route-circle.py
class Solution(object):
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        
        path = {"R": [1, 0], "L": [-1, 0], "U": [0, 1], "D": [0, -1]}
        
        x, y = 0, 0
        
        for m in moves:
            x += path[m][0]
            y += path[m][1]
        
        return x == 0 and y == 0

if __name__ == '__main__':
    s = Solution()

    print s.judgeCircle("RULD")
    print s.judgeCircle("RULR")
